smartsim/_core/launcher/dragon/dragonSockets.py

```diff
@@ -115,20 +117,28 @@ def get_authenticator(
     :type context: zmq.Context
     :returns: the activated `Authenticator`
     :rtype: zmq.auth.thread.ThreadAuthenticator"""
+    # pylint: disable-next=global-statement
+    global AUTHENTICATOR
+
+    if AUTHENTICATOR is not None:
+        if AUTHENTICATOR.is_alive():
+            return AUTHENTICATOR
+        del AUTHENTICATOR
```

Comment:
would `AUTHENTICATOR.stop()` here be a good idea?

Reply:
Yeah, totally.